optimum/amd/ryzenai/pipelines/__init__.py

```diff
@@ -0,0 +1,196 @@
+}
+
+RYZENAI_SUPPORTED_TASKS = {
+    "image-classification": {
+        "impl": ImageClassificationPipeline,
+        "class": (RyzenAIModelForImageClassification,),
+        "default": "amd/resnet50",
+        "type": "image",
+    },
+    "object-detection": {
+        "impl": YoloObjectDetectionPipeline,
+        "class": (RyzenAIModelForObjectDetection,),
+        "default": "amd/yolox-s",
+        "type": "image",
+        "model_type": "yolox",
+    },
+}
+
+
+def load_model(
+    model,
+    task,
+    model_type,
+    vaip_config,
+    SUPPORTED_TASKS,
+    token: Optional[Union[bool, str]] = None,
+    revision: str = "main",
+):
+    if model is None:
+        if task != "object-detection":
+            raise ValueError("Creating pipeline without model for the task is not supported!")
+
+        model_id = SUPPORTED_TASKS[task]["default"]
+        if model_type is None:
+            model_type = SUPPORTED_TASKS[task].get("model_type", None)
+        model = SUPPORTED_TASKS[task]["class"][0].from_pretrained(
+            model_id, vaip_config=vaip_config, use_auth_token=token, revision=revision
+        )
+    elif isinstance(model, str):
+        model_id = model
+        ort_model_class = SUPPORTED_TASKS[task]["class"][0]
+
+        model = ort_model_class.from_pretrained(
+            model_id, vaip_config=vaip_config, use_auth_token=token, revision=revision, provider="CPUExecutionProvider"
+        )
+    elif isinstance(model, RyzenAIModel):
+        model_id = None
+    else:
+        raise ValueError(
+            f"Model {model} is not supported. Please provide a valid model either as string or RyzenAIModel."
+            "You can also provide non model then a default one will be used"
+        )
+    return model, model_id, model_type
+
+
+def pipeline(
+    task,
+    vaip_config,
+    model: Optional[Any] = None,
+    model_type: str = None,
+    feature_extractor: Optional[Union[str, "PreTrainedFeatureExtractor"]] = None,
+    image_processor: Optional[Union[str, BaseImageProcessor]] = None,
+    use_fast: bool = True,
+    token: Optional[Union[str, bool]] = None,
+    revision: Optional[str] = None,
+    **kwargs,
+) -> Pipeline:
+    """
+    Utility method to build a pipeline for various RyzenAI tasks.
+
+    This function creates a pipeline for a specified task, utilizing a given model or loading the default model for the task.
+    The pipeline includes components such as a image processor and model.
+
+    Args:
+        task (str):
+            The task defining which pipeline will be returned. Available tasks include:
+            - "image-classification"
+            - "object-detection"
+
+        vaip_config (str):
+            Runtime configuration file for inference with Ryzen IPU. A default config file can be found in the Ryzen AI VOE package,
+            extracted during installation under the name `vaip_config.json`.
+
+        model (Optional[Any], *optional*):
```

Comment:
We decided not to use `*optional*` in optimum libs, let's stick to it https://huggingface.slack.com/archives/C02P0559X9S/p1669628754896019